### ISSUE TITLE
riscv-arch-test PMP tests not yet passing

### ISSUE THREAD

**Issue by davidharrishmc** [openhwgroup/cvw#1211]
wsim rv32gc arch32pmp
```
   Error on test rv32i_m/pmp32/src/pmp-CFG-reg.S result           1: adr = 80004514 sim (D$) ffffffff signature = 9f9f9f9f
```

**Comment by davidharrishmc** [comment]
Bits 6:5 of each PMPcfg should be WARL 00, but appear to be writeable in Wally.

**Comment by davidharrishmc** [comment]
Fixed with PR #1212.

However, RV64 PMP is failing.
```
harris@chips:~/cvw$ wsim --sim verilator  rv64gc arch64pmp
rv64i_m/pmp64/src/pmp64-CFG-reg.S succeeded.  Brilliant!!!
rv64i_m/pmp64/src/pmp64-CSR-access.S succeeded.  Brilliant!!!
  Error on test rv64i_m/pmp64/src/pmp64-NA4-R-priority-level-2.S result           3: adr = 0000000080004228 sim (D$) 0000001300148493 signature = 0000000000148493
```

**Comment by davidharrishmc** [comment]
ld from 80000828 is mismatching Imperas at 0x800005e0.  ref has no access permissions.
Looks like a ld, in which pmpcfg0 has permissions related to 828-82b, only for one word, not double word.

**Comment by davidharrishmc** [comment]
In corm, PMPCFGreg should take CSRPMPLegalizedWriteValM rather than CSRPMPWriteValM & PMPCFG_MASK.

**Comment by davidharrishmc** [comment]
Fixed size of access failure above with PR #1212.

Now seeing 
#   Error on test rv64i_m/pmp64/src/pmp64-NAPOT-RWX.S result          34: adr = 0000000080004320 sim (D$) deadbeefdeadbeef signature = 00000000000010e3

**Comment by davidharrishmc** [comment]
This is due to executing an instruction at 80000820.  The dut has the original instruction at that address 00148493 addi x9, x9, 1, while the ref has a modified instruction 00000013 addi x0, x0, 0.

The 00000013 gets written to 80000820.  However, there doesn't seem to be a fence.i.

This presumably affects other tests, but maybe the signature isn't propagated out.

Adding fence.i to pmp64-NAPOT-RWX.S and pmp64-TOR-RWX.S before the jal in VERIFICATION_RWX (and adding Zifencei to RVTEST_ISA nd RVTEST_CASE) fixed these and now all PMP tests pass.
